core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -61,17 +65,22 @@ class CityConstructions {
         toReturn.currentConstructionIsUserSet = currentConstructionIsUserSet
         toReturn.constructionQueue.addAll(constructionQueue)
         toReturn.productionOverflow = productionOverflow
+        toReturn.freeBuildingsProvidedFromThisCity.putAll(freeBuildingsProvidedFromThisCity)
         return toReturn
     }
 
     internal fun getBuildableBuildings(): Sequence<Building> = cityInfo.getRuleset().buildings.values
-            .asSequence().filter { it.isBuildable(this) }
+        .asSequence().filter { it.isBuildable(this) }
 
     fun getConstructableUnits() = cityInfo.getRuleset().units.values
-            .asSequence().filter { it.isBuildable(this) }
+        .asSequence().filter { it.isBuildable(this) }
 
     fun getBasicCultureBuildings() = cityInfo.getRuleset().buildings.values
-            .asSequence().filter { it.culture > 0f && !it.isAnyWonder() && it.replaces == null }
+        .asSequence().filter { it.culture > 0f && !it.isAnyWonder() && it.replaces == null }
+    
+    fun getBasicStatBuildings(stat: Stat) = cityInfo.getRuleset().buildings.values
+        .asSequence()
+        .filter { !it.isAnyWonder() && it.replaces == null && it.getStats(cityInfo)[stat] > 0f }
```

Comment:
`it[stat] > 0f` or this is no longer "basic"? You get a Belief that adds Culture to a Shrine and this function would return the Shrine as Culture building. If intentional - Kdoc what & why.

Reply:
Good point, didn't think about that, fixed

Reply:
Except there is no such Belief - but for Temples :rofl: